packages/server/src/main/java/software/uncharted/terarium/hmiserver/controller/dataservice/ChartAnnotationController.java

```diff
@@ -0,0 +1,175 @@
+import org.springframework.http.MediaType;
+import org.springframework.http.ResponseEntity;
+import org.springframework.security.access.annotation.Secured;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
+import org.springframework.web.bind.annotation.RequestBody;
+import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
+import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.server.ResponseStatusException;
+import software.uncharted.terarium.hmiserver.models.dataservice.ChartAnnotation;
+import software.uncharted.terarium.hmiserver.models.dataservice.ResponseDeleted;
+import software.uncharted.terarium.hmiserver.security.Roles;
+import software.uncharted.terarium.hmiserver.service.CurrentUserService;
+import software.uncharted.terarium.hmiserver.service.data.ChartAnnotationService;
+import software.uncharted.terarium.hmiserver.service.data.ProjectAssetService;
+import software.uncharted.terarium.hmiserver.service.data.ProjectService;
+import software.uncharted.terarium.hmiserver.utils.rebac.Schema;
+
+@RequestMapping("/chart-annotations")
+@RestController
+@Slf4j
+@RequiredArgsConstructor
+@Transactional
+public class ChartAnnotationController {
+
+	final ChartAnnotationService chartAnnotationService;
+
+	final ProjectAssetService projectAssetService;
+
+	final ProjectService projectService;
+
+	final CurrentUserService currentUserService;
+
+	private static class SearchRequestBody {
+
+		public UUID nodeId;
+	}
+
+	@PostMapping("/search")
+	@Secured(Roles.USER)
+	@Operation(summary = "Gets a list of chart annotations by provided node ID")
+	@ApiResponses(
+		value = {
+			@ApiResponse(
+				responseCode = "200",
+				description = "chart annotations found.",
+				content = @Content(
+					mediaType = MediaType.APPLICATION_JSON_VALUE,
+					schema = @io.swagger.v3.oas.annotations.media.Schema(implementation = ChartAnnotation.class)
+				)
+			),
+			@ApiResponse(responseCode = "204", description = "There was no chart annotation found", content = @Content),
+			@ApiResponse(
+				responseCode = "500",
+				description = "There was an issue retrieving the chart annotations from the data store",
+				content = @Content
+			)
+		}
+	)
+	public ResponseEntity<List<ChartAnnotation>> getChartAnnotationsByNodeId(
+		@RequestParam(name = "project-id", required = false) final UUID projectId,
+		@RequestBody final SearchRequestBody body
+	) {
+		final Schema.Permission permission = projectService.checkPermissionCanRead(
+			currentUserService.get().getId(),
+			projectId
+		);
+
+		final List<ChartAnnotation> chartAnnotations = chartAnnotationService.getAnnotationsByNodeId(
+			body.nodeId,
+			permission
+		);
+
+		if (chartAnnotations.isEmpty()) {
+			return ResponseEntity.noContent().build();
+		}
```

Comment:
This feels a little bit weird - wouldn't we want to return an empty array instead of non-content 204? 
Are we doing this across the app @dvince2 ?

That way the client-side doesn't need to have two different branches dealing with it.

Reply:
Hm.. yeah I think it's preferable to return an empty array instead of 204. I just followed code from other examples like summary endpoint. If there's no specific reason to return 204 I'll just updated it to return an empty array if it's empty.

Reply:
We do return `no content` other places, but there isn't a set standard that the front end works with.